### ISSUE TITLE
Add date-based paging support to EndpointPaging

### ISSUE THREAD

**Issue by rinukkusu** [rinukkusu/spotify-dart#75]
In PR #66 date-based paging has been added to the `me.recentlyPlayed()` endpoint.
As a developer I'd like to use this kind of paging in other places as well, so it would be great to have it implemented in [`EndpointPaging`](https://github.com/rinukkusu/spotify-dart/blob/master/lib/src/endpoints/endpoint_paging.dart) itself directly. We'll probably need seperate paging classes for offset- and date-based paging.

**Comment by hayribakici** [comment]
It would be good to know, which API endpoints are providing the `before` or `after` query parameters. So far, I have seen it only in the `recently-played` endpoint. Do you know any other?

**Comment by rinukkusu** [comment]
True, I have not yet investigated which endpoints are actually using that. If we find out, that `recently-played` is the only one - great, then we can close the PR 😄

**Comment by thosakwe** [comment]
Were there any other endpoints that use date-based paging

**Comment by hayribakici** [comment]
I looked it into today and so far only the `recenty-played` endpoint has a timestamp (`before` and `after`) query parameters.
